app/src/androidTest/java/com/github/se/endToEnd/SettingsFlowEndToEnd.kt

```diff
@@ -0,0 +1,293 @@
+import androidx.compose.ui.test.onNodeWithText
+import androidx.compose.ui.test.performClick
+import androidx.compose.ui.test.performScrollTo
+import androidx.compose.ui.test.performTextClearance
+import androidx.compose.ui.test.performTextInput
+import androidx.core.app.ActivityCompat
+import androidx.test.core.app.ActivityScenario
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import com.github.se.icebreakrr.MainActivity
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.firestore.FirebaseFirestore
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import javax.inject.Inject
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+@HiltAndroidTest
+class SettingsFlowEndToEnd {
+  private val intent =
+      Intent(InstrumentationRegistry.getInstrumentation().targetContext, MainActivity::class.java)
+
+  @Inject lateinit var authInjected: FirebaseAuth
+  @Inject lateinit var firestoreInjected: FirebaseFirestore
+  @Inject lateinit var authStateListener: FirebaseAuth.AuthStateListener
+
+  @get:Rule var hiltRule = HiltAndroidRule(this)
+
+  @get:Rule val composeTestRule = createComposeRule()
+
+  @Before
+  fun setup() {
+    hiltRule.inject()
+    intent.putExtra("IS_TESTING", true)
+    ActivityCompat.setPermissionCompatDelegate(TestPermissionDelegate())
+  }
+
+  @Test
+  fun SettingsFlowEndToEndTest() {
+    ActivityScenario.launch<MainActivity>(intent).use { scenario ->
+      // Screen 1 : Around You Screen
+      composeTestRule.onNodeWithTag("aroundYouScreen").assertIsDisplayed()
+      composeTestRule.onNodeWithTag("bottomNavigationMenu").assertIsDisplayed()
+      composeTestRule.waitForIdle()
+      composeTestRule.onAllNodesWithTag("profileCard").onFirst().assertIsDisplayed()
+      composeTestRule.onNodeWithTag("topBar").assertIsDisplayed()
+      composeTestRule.onNodeWithTag("filterButton").assertIsDisplayed()
+
+      // go to settings :
+      composeTestRule.onNodeWithText("Settings").performClick()
+
+      // check if everything is displayed :
+      composeTestRule.onNodeWithTag("profileCard").assertIsDisplayed()
+      composeTestRule.onNodeWithTag("settingsScreen").assertIsDisplayed()
+      composeTestRule.onNodeWithTag("logOutButton").assertIsDisplayed().assertHasClickAction()
+      composeTestRule.onNodeWithTag("Toggle Discoverability").assertIsDisplayed()
+
+      // click on your profile :
+      composeTestRule.onNodeWithTag("profileCard").performClick()
+
+      // test if everything is displayed
+      composeTestRule.onNodeWithTag("profileScreen").assertIsDisplayed()
+      composeTestRule.onNodeWithTag("infoSection").assertIsDisplayed()
+      composeTestRule.onNodeWithTag("profileHeader").assertIsDisplayed()
+      composeTestRule.onNodeWithTag("profilePicture").assertIsDisplayed()
```

Comment:
Thank you for the tip! I didn't know that you could create a function on an object that you didn't implement.  But I don't think it would be that modular because if I do what you propose, I will go from : 
```
composeTestRule.onNodeWithTag("profileScreen").assertIsDisplayed()
composeTestRule.onNodeWithTag("infoSection").assertIsDisplayed()
composeTestRule.onNodeWithTag("profileHeader").assertIsDisplayed()
 composeTestRule.onNodeWithTag("profilePicture").assertIsDisplayed()
```
to 
```
composeTestRule.verifyDisplayed("profileScreen")
composeTestRule.verifyDisplayed("infoSection")
composeTestRule.verifyDisplayed("profileHeader")
composeTestRule.verifyDisplayed("profilePicture")
```
which isn't that modular I think. Also if I do that I would need to make a lot of different functions because sometimes I check if it is displayed, then I check if it has click action and then I click on it, and sometimes I check that it is displayed, that is has click action but I don't click on it. There are a lot of different scenarios that would need one different function each.